suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -0,0 +1,165 @@
+import { useSelector } from 'react-redux';
+import Animated, {
+    useAnimatedStyle,
+    useSharedValue,
+    withDelay,
+    withTiming,
+} from 'react-native-reanimated';
+import { useEffect, useState } from 'react';
+
+import { useNavigation } from '@react-navigation/native';
+import { A } from '@mobily/ts-belt';
+
+import {
+    ConnectDeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import { getSortedDevicesWithoutInstances } from '@suite-common/suite-utils';
+import { analytics, EventType } from '@suite-native/analytics';
+import {
+    DeviceRootState,
+    PORTFOLIO_TRACKER_DEVICE_ID,
+    selectDevice,
+    selectDeviceById,
+    selectPhysicalDevices,
+} from '@suite-common/wallet-core';
+import { Button, HStack, VStack, Text, Divider, Box } from '@suite-native/atoms';
+import { AcquiredDevice, TrezorDevice } from '@suite-common/suite-types';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { DeviceItem } from './DeviceItem';
+import { useDeviceManager } from '../hooks/useDeviceManager';
+import { MANAGER_MODAL_BOTTOM_RADIUS } from './DeviceManagerModal';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    RootStackParamList,
+    RootStackRoutes.AppTabs,
+    RootStackParamList
+>;
+
+type DeviceListProps = {
+    isVisible: boolean;
+    onSelectedDevice?: (device: AcquiredDevice) => void;
+};
+
+const ITEM_HEIGHT = 62;
+const SPACING = 8;
+const BUTTON_HEIGHT = 48;
+const SEPARATOR_HEIGHT = 18;
+const VERTICAL_PADDING = 16;
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.medium,
+}));
+
+export const DeviceList = ({ isVisible, onSelectedDevice }: DeviceListProps) => {
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+    const { setIsDeviceManagerVisible } = useDeviceManager();
+    const device = useSelector(selectDevice);
+    const physicalDevices = useSelector(selectPhysicalDevices);
+    const portfolioDevice = useSelector((state: DeviceRootState) =>
+        selectDeviceById(state, PORTFOLIO_TRACKER_DEVICE_ID),
+    );
+    const { utils } = useNativeStyles();
+    const opacity = useSharedValue(0);
+    const height = useSharedValue(0);
+    const [isShown, setIsShown] = useState(false);
```

Comment:
Why is `isShown` needed at all? Does not it just reflect the prop value of `isVisible`?

Reply:
This is needed so the height animation of removal finishes correctly and list stops rendering just after it finished

Reply:
In that case, I believe that you should use [entering/exiting](https://docs.swmansion.com/react-native-reanimated/docs/layout-animations/entering-exiting-animations) animation and remove the `isShown` state. 

It is even possible to create a custom entering/exiting animation. [See this.](https://docs.swmansion.com/react-native-reanimated/docs/layout-animations/custom-animations)

Reply:
I've tried that. But it is suboptimal, because the element size goes away immediately and walletList (that is below) jumps up right away not waiting for animation to finish.

Reply:
@juriczech Did not you have simillar problem? Were you able to solve it?